quipucords/api/common/entities.py

```diff
@@ -21,18 +24,46 @@
 from compat.db import StringAgg
 from constants import DataSources
 
-CANONICAL_FACTS = (
-    "fqdn",
-    "bios_uuid",
-    "insights_id",
-    "ip_addresses",
+logger = getLogger(__name__)
+
+
+"""
+List of fact names for synthesizing an alternate provider_id value.
+
+These facts should be generally unique enough to differentiate two hosts but
+also be static enough *not* to cause the same real-world host to appear as
+different hosts when running two scans over a short period of time.
+"""
+FINGERPRINT_FACT_NAMES_TO_SYNTHESIZE_PROVIDER_ID = {
+    "name",
+    "os_name",
+    "infrastructure_type",
+    "cloud_provider",
     "mac_addresses",
-    "satellite_id",
+    "ip_addresses",
```

Comment:
I would personally drop ip_addresses from what we use to create the provider_id, that can be updated by DNS, they unplug the machine one place and plug it somewhere else, different provider_id.  I think the key is to use hardware/static type data, i.e. mac_addresses, etc.

Reply:
I have no strong opinion on this one. On one hand, you are correct that IP addresses are mutable. On the other hand, if a system changes IP addresses but doesn't have an `/etc/machine_id`, is it even _the same system_ for counting purposes? We have a minor Ship of Theseus problem here. I'll gather feedback in Slack before making changes here.

Reply:
Due to the time restrictions, this is ok for now. We can refine/fine-tune with a later PR.